components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/dao/JDBCPolicyDAOImpl.java

```diff
@@ -0,0 +1,1560 @@
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_ALL_PAP_POLICIES_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_ALL_PDP_POLICIES_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_LATEST_POLICY_VERSION_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_PAP_POLICY_BY_VERSION_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_PAP_POLICY_EDITOR_DATA_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_PAP_POLICY_IDS_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_PAP_POLICY_META_DATA_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_PAP_POLICY_REFS_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_PAP_POLICY_SET_REFS_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_PAP_POLICY_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_PDP_POLICY_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_POLICY_PAP_PRESENCE_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_POLICY_PDP_PRESENCE_BY_VERSION_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_POLICY_PDP_PRESENCE_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_POLICY_VERSIONS_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_PUBLISHED_POLICY_VERSION_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.PUBLISH_POLICY_VERSION_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.RESTORE_ACTIVE_STATUS_AND_ORDER_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.UPDATE_ACTIVE_STATUS_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.UPDATE_ORDER_SQL;
+
+import static java.time.ZoneOffset.UTC;
+
+/**
+ * This class handles the policy operations in the JDBC data store.
+ */
+public class JDBCPolicyDAOImpl extends AbstractPolicyFinderModule implements PolicyDAO {
+
+    private static final Log LOG = LogFactory.getLog(JDBCPolicyDAOImpl.class);
+    private static final String MODULE_NAME = "JDBC Policy Finder Module";
+    private static final String ERROR_RETRIEVING_PAP_POLICY =
+            "Error while retrieving entitlement policy %s from the PAP policy store";
+    private static final String ERROR_RETRIEVING_POLICIES_FROM_POLICY_FINDER = "Policies can not be retrieved from " +
+            "the policy finder module";
+    private final int maxVersions;
+    private static final String POLICY_ID_1 = "POLICY_ID_1";
```

Comment:
Lets validate whether using `setString()` twice would avoid the need of a different key.

Reply:
This is what was observed:
- When we are setting the same value to two variables, it sets correctly
- When we want to set two different values, it still sets the same value to both variables, in this case the latter value is set.
 ```
            demotePolicyPrepStmt.setBoolean(IS_IN_PDP, !IN_PDP);
            demotePolicyPrepStmt.setBoolean(IS_ACTIVE, INACTIVE);
            demotePolicyPrepStmt.setInt(POLICY_ORDER, DEFAULT_POLICY_ORDER);
            demotePolicyPrepStmt.setBoolean(IS_IN_PDP, IN_PDP);
            demotePolicyPrepStmt.setString(POLICY_ID, policyId);
            demotePolicyPrepStmt.setInt(TENANT_ID, tenantId);
```

Reply:
Avoided setting the values twice where the same value is being set.